src/libdredd/src/mutate_ast_consumer.cc

```diff
@@ -108,37 +108,7 @@ void MutateAstConsumer::HandleTranslationUnit(clang::ASTContext& ast_context) {
     return;
   }
 
-  // Rewrite the size expressions of constant-sized arrays as needed.
-  for (const auto& constant_sized_array_decl :
-       visitor_->GetConstantSizedArraysToRewrite()) {
-    assert(compiler_instance_->getLangOpts().CPlusPlus &&
-           constant_sized_array_decl->getType()->isConstantArrayType());
-    auto typeloc = constant_sized_array_decl->getTypeSourceInfo()->getTypeLoc();
-
-    // Note: in some cases a declaration with constant array type does not have
-    // an associated ConstantArrayTypeLoc object. This happens, for example,
-    // when structured bindings are used. For example, consider:
-    //
-    //     auto [x, y] = a;
-    //
-    // This yields a constant-sized array, but there is no explicit array type
-    // declaration. In such cases, no action is required.
-    //
-    // We use a loop to descend into nests of arrays.
-    while (
-        auto constant_array_typeloc =
-            typeloc.getUnqualifiedLoc().getAs<clang::ConstantArrayTypeLoc>()) {
-      RewriteExpressionInMainFileToIntegerConstant(
-          constant_array_typeloc.getSizeExpr(),
-          llvm::dyn_cast<clang::ConstantArrayType>(
-              constant_array_typeloc.getType()
-                  ->getAsArrayTypeUnsafe()
-                  ->getAsArrayTypeUnsafe())
-              ->getSize()
-              .getLimitedValue());
-      typeloc = constant_array_typeloc.getElementLoc();
-    }
-  }
+  RewriteExpressionInMainFile();
```

Comment:
I suggest using a plural - "RewriteExpressionsInMainFile" since it may be more than one expression.